### PR TITLE
Adding seperate parameters for the subnets #54764

### PR DESCRIPTION
**Pull request by abika5**

### CODE REVIEW

File: mlcluster.template [hunk 1]
```diff
@@ -383,7 +407,7 @@ Resources:
         S3Directory: !FindInMap [Variable,"S3Directory","base"]
         NodesPerZone: !Ref NodesPerZone
         NumberOfZones: !Ref NumberOfZones
-        Subnets: !If [MultiZone, !Join [',', !Ref PrivateSubnets], !Select [0, !Ref PrivateSubnets]]
+        Subnets: !If [MultiZone, !Join [',', [!Ref PrivateSubnet1, !Ref PrivateSubnet2, !Ref PrivateSubnet3]], !Ref PrivateSubnet1]
```
Comment (vzhang-progress) [inline mlcluster.template:410]: Does Fn::Join work with empty value? Since subnet2 and 3 may be empty.

Thanks,
Vanessa

Comment (mattsunsjf) [reply mlcluster.template:410]: No.

Comment (abika5) [reply mlcluster.template:410]: Yes, the Join won't work for the empty values. This is why we have the MultiZone condition. This condition is true when we provide 3 AZs. So as the mapping goes, when multizone is true we use join. If it is false, then we use the First private/public subnet

MultiZone:
    !Not [!Equals [!Ref NumberOfZones, 1]]

**Comment by abika5** [comment]
I have found another issue for this change. I am going to contact the AWS support and get back to this PR. We might have more changes.

**Review by mattsunsjf — APPROVED** [review]

**Review by itsshivaverma — APPROVED** [review]

**Review by sebastiankalinowski — APPROVED** [review]

**Review by vzhang-progress — APPROVED** [review]

**Pull request merged**